addons/payment_razorpay/models/payment_transaction.py

```diff
@@ -75,14 +75,14 @@ def _validate_phone_number(self, phone):
         :raise ValidationError: If the phone number is missing or incorrect.
         """
         if not phone:
-            raise ValidationError("Razorpay: " + _("The phone number is missing."))
+            return ""
 
         try:
             phone = self._phone_format(
                 number=phone, country=self.partner_country_id, raise_exception=True
             )
         except Exception:
-            raise ValidationError("Razorpay: " + _("The phone number is invalid."))
+            return ""
```

Comment:
Same as above
Also you should at least log exception or warning to display it in terminal I guess

Reply:
Also since we do not check if phone is set or not and we are expecting user to add proper phone and if user don't have proper phone then user will update it from razorpay form then
shouldn't we remove that('validate_phone_number') method completely and add directly `self.partner_phone or ""` or do we need to format phone is added because  even though there is exception in formatting you still pass `""` so IMO better to add dirty phone number instead of empty string
Don't you think?

Reply:
@kcv-odoo If the invalid phone is passed then Razorpay give an error.
but if we pass `""` then no error and ask phone number from the user when the popup is open
so still we need the method `validate_phone_number`